tests/data_context/test_base_data_context.py

```diff
@@ -161,55 +168,155 @@ def test_get_config_with_variables_substituted(
+    monkeypatch.setenv("MY_PLUGINS_DIRECTORY", "plugins/")
+    context = DataContext(ge_cloud_mode=True)
 
-    context.get_expectation_suite("my_expectation_suite")
-    assert mock_cloud_get_expectation_suite.call_count == 1
+    (
+        validator,
+        _,
+    ) = prepare_validator_for_cloud_e2e(context)
+
+    with mock.patch("requests.put", autospec=True) as mock_put:
+        type(mock_put.return_value).status_code = mock.PropertyMock(return_value=200)
+        validator.save_expectation_suite()
+
+    assert mock_put.call_count == 1
 
 
 @pytest.mark.cloud
-@mock.patch("great_expectations.data_context.CloudDataContext.delete_expectation_suite")
-def test_delete_expectation_suite_with_cloud_enabled_context_uses_cloud_impl(
-    mock_cloud_delete_expectation_suite: mock.MagicMock,
-    empty_cloud_data_context: BaseDataContext,
+@pytest.mark.integration
+def test_validator_e2e_workflow_with_cloud_enabled_context(
+    monkeypatch,
+    prepare_validator_for_cloud_e2e: Callable[[DataContext], Tuple[Validator, str]],
 ) -> None:
     """
     What does this test do and why?
 
-    Ensures that when a BaseDataContext is instantiated with cloud_mode=True, the call to
-    ExpectationSuite CRUD leverages the implementation defined and managed by CloudDataContext.
+    Ensures that the Validator that is created through DataContext.get_validator() is
+    Cloud-enabled if the DataContext used to instantiate the object is Cloud-enabled.
+    Saving of ExpectationSuites using such a Validator should send payloads to the Cloud
+    backend.
     """
-    context = empty_cloud_data_context
+    # Context is shared between other tests so we need to set a required env var
+    monkeypatch.setenv("MY_PLUGINS_DIRECTORY", "plugins/")
+    context = DataContext(ge_cloud_mode=True)
+
+    (
+        validator,
+        expectation_suite_ge_cloud_id,
+    ) = prepare_validator_for_cloud_e2e(context)
 
-    context.delete_expectation_suite("my_expectation_suite")
-    assert mock_cloud_delete_expectation_suite.call_count == 1
+    assert len(validator.expectation_suite.expectations) == 4
+
+    res = validator.expect_column_max_to_be_between("x", min_value=0, max_value=10)
+    assert res.success
+
+    assert len(validator.expectation_suite.expectations) == 5
+
+    # Confirm that the suite is present before saving
+    assert expectation_suite_ge_cloud_id in context.list_expectation_suite_names()
+    suite_on_context = context.get_expectation_suite(
+        ge_cloud_id=expectation_suite_ge_cloud_id
+    )
+    assert str(expectation_suite_ge_cloud_id) == str(suite_on_context.ge_cloud_id)
+
+    validator.save_expectation_suite()
+    assert len(validator.expectation_suite.expectations) == 5
+
+    # Confirm that the suite is present after saving
+    assert expectation_suite_ge_cloud_id in context.list_expectation_suite_names()
+    suite_on_context = context.get_expectation_suite(
+        ge_cloud_id=expectation_suite_ge_cloud_id
```

Comment:
Maybe one last check to make sure the retrieved suite (`suite_on_context`) still has 5 expectations?

Reply:
done!